### ISSUE TITLE
🐛 `biome lint --only` does not filter out `suppressions/incorrect`

### ISSUE THREAD

**Issue by Loskir** [biomejs/biome#6523]
### Environment information

```block
CLI:
  Version:                      2.0.5
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.9.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

1. Create a file
```ts
// biome-ignore lint/complexity/useDateNow: <explanation>
const a = +new Date();
```
2. Run `biome lint --only correctness/noUnusedVariables`
3. It outputs both `lint/correctness/noUnusedVariables` and `suppressions/incorrect`

### Expected result

It should output only `lint/correctness/noUnusedVariables`

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Can you tell us what `suppression/incorrect` contains? What's it complaining about? Can you share the whole message with us?

**Comment by Loskir** [comment]
```
test2.ts:1:45 suppressions/incorrect ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ A suppression shouldn't have an <explanation> placeholder. Example of suppression: // biome-ignore lint: false positive
  
  > 1 │ // biome-ignore lint/complexity/useDateNow: <explanation>
      │                                             ^^^^^^^^^^^^^
    2 │ const a = +new Date();
    3 │ 
  

test2.ts:2:7 lint/correctness/noUnusedVariables  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This variable a is unused.
  
    1 │ // biome-ignore lint/complexity/useDateNow: <explanation>
  > 2 │ const a = +new Date();
      │       ^
    3 │ 
  
  ℹ Unused variables are often the result of an incomplete refactoring, typos, or other sources of bugs.
  
  ℹ Unsafe fix: If this is intentional, prepend a with an underscore.
  
    1 1 │   // biome-ignore lint/complexity/useDateNow: <explanation>
    2   │ - const·a·=·+new·Date();
      2 │ + const·_a·=·+new·Date();
    3 3 │   
  

Checked 3 files in 6ms. No fixes applied.
Found 1 error.
Found 1 warning.
lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

**Comment by ematipico** [comment]
That particular suppression and message are expected considering your use case. That's a warning about the fact that the suppression text and format aren't syntactically correct (it doesn't check the rule).

**Comment by Loskir** [comment]
Yes, but this message (`suppression/incorrect`) is unrelated to the issue i'm filtering for (`lint/correctness/noUnusedVariables`) and I would expect not to see it when I'm explicitly filtering for a certain rule

**Comment by ematipico** [comment]
Not really, the message isn't about the rule, but about that the suppression is syntactically incorrect. If you change `<explanation>` to something else, you shouldn't see that anymore.

Instead, if you see another message, then I suppose there's a bug there.

**Comment by Loskir** [comment]
I get your idea. The suppression itself works (i.e. suppresses `useDateNow`) but has an invalid explanation, I completely agree with that.

My point is that with `--only` it shouldn't be shown. It's not a critical issue (merely a warning) and I don't like the fact that it bypasses `--only`

**Comment by Androlax2** [comment]
Had the same issue

**Comment by ematipico** [comment]
> My point is that with --only it shouldn't be shown. It's not a critical issue (merely a warning) and I don't like the fact that it bypasses --only

That's because you link the suppression to the rule, while what I'm trying to say is that there's no connection. The suppression is syntactically incorrect, hence Biome emits a diagnostic. You could change the name of the rule, or even add `lint` as a category, and you'd still see the same message
